shared/js/background/chrome-events.es6.js

```diff
@@ -0,0 +1,182 @@
+/**
+ * NOTE: this needs to be the first listener that's added
+ *
+ * on FF, we might actually miss the onInstalled event
+ * if we do too much before adding it
+ */
+const ATB = require('./atb.es6')
+
+chrome.runtime.onInstalled.addListener(function(details) {
+    // only run the following section on install and on update
+    if (details.reason.match(/install|update/)) {
+        ATB.updateATBValues();
+    }
+
+    if (details.reason.match(/install/)) {
+        ATB.openPostInstallPage()
+    }
+})
+
+/**
+ * REQUESTS
+ */
+
+const constants = require('../../data/constants')
+const redirect = require('./redirect.es6')
+const tabManager = require('./tab-manager.es6')
+
+chrome.webRequest.onBeforeRequest.addListener(
+    redirect.handleRequest,
+    {
+        urls: ['<all_urls>'],
+        types: constants.requestListenerTypes
+    },
+    ['blocking']
+)
+
+chrome.webRequest.onHeadersReceived.addListener(
+    (request) => {
+        if (request.type === 'main_frame') {
+            tabManager.updateTabUrl(request)
+        }
+
+        if (/^https?:\/\/(.*?\.)?duckduckgo.com\/\?/.test(request.url)) {
+            // returns a promise
+            return ATB.updateSetAtb(request)
+        }
```

Comment:
Merged these into the same handler. No idea how to test this manually though...

Reply:
We'll need to double check this when we have a beta version to test through the ATB modal.